### DISCUSSION TITLE
Feature Request: Add 'Enter' and 'Leave' Types to NotebookEditorSelectionChangeEvent

### DISCUSSION THREAD

**Discussion by liorshk** [microsoft/vscode-discussions#1911] [Extension Development QnA] [answered]
I’m encountering an issue where the NotebookEditorSelectionChangeEvent triggers twice, making it challenging to determine the current selection. Specifically, it seems that the event triggers both for the change "from" a selection and "to" a selection, but without clear differentiation between the two.

Would it be possible to add event types like "leave" or "enter" to better identify the context of each selection change?

**Comment by isidorn** [comment]
@rebornix might be able to help answer this

**Comment by rebornix** [comment, marked as answer]
Let's track this in https://github.com/microsoft/vscode/issues/237913
